### PR TITLE
utils.ping_tcp_port: Support remote endpoints and also IPv6

### PR DESCRIPTION
**Pull request by andir**
Previously only IPv4 was supported. `socket.getaddrinfo` is used to
iterate a list of protocols and endpoints of the given `host`
(previsouly just `ip`).

### CODE REVIEW

**Comment by andir** [comment]
What do you mean with `otherwise`? Is there anything I should change?

File: nixops/util.py [hunk 1]
```diff
@@ -148,22 +148,32 @@ def make_non_blocking(fd):
...
+           continue
+        except (ConnectionError, OSError):
+            # Reset, Refused, Aborted, No route to host
+            if ensure_timeout: time.sleep(timeout)
+            # continue with the next address
+            continue
+        except:
+            raise
+        else:
+          s.shutdown(socket.SHUT_RDWR)
```
Comment (globin) [inline nixops/util.py]: Indentation is off here

File: nixops/util.py [hunk 2]
```diff
@@ -148,22 +148,32 @@ def make_non_blocking(fd):
...
+    infos = socket.getaddrinfo(host, port, 0, 0, socket.IPPROTO_TCP)
+    for info in infos:
+        s = socket.socket(info[0], info[1])
+        s.settimeout(timeout)
+        s.setsockopt(socket.SOL_SOCKET, socket.SO_LINGER, struct.pack('ii', 1, 0))
+        try:
+            s.connect(info[4])
+        except socket.timeout:
+           # try next address
+           continue
```
Comment (globin) [inline nixops/util.py]: Indentation is off here

**Comment by andir** [comment]
fixed

**Pull request merged**